### DISCUSSION TITLE
Custom rule

### DISCUSSION THREAD

**Discussion by rambant777** [wazuh/wazuh#27284] [Q&A]
Hi!
I created a rule that should trigger if a user is added to the “IT-adm” group.
In wazuh-logtest I put the code { “win”: { “system”: { “eventID”: “4728” }, “eventdata”: { “targetUserName”: “*IT-adm*” } } } }, and I don't see the rule triggering because there is no Phase 3. How to fix the situation?
```
  <rule id="100086" level="10">
    <if_sid>18125</if_sid>
    <field name="win.system.eventID">4728|4732</field>
    <regex field="data.win.eventdata.targetUserName">IT-adm</regex>
    <description>Add user in group IT-adm</description>
    <options>no_full_log</options>
    <group>active_directory,group_changes</group>
   </rule>
</group>

```

**Comment by matias-braida** [comment]
Hello rambant777,

The log string you are feeding to the "wazuh-logtest" tool seems to have an incorrect format.

The "wazuh-logtest" tool works with the log string in the same format the wazuh manager receives them. Here is one example:
```
Sep 26 15:55:59 VBox su: (to root) sudo on pts/2
```

To get the log string with the correct format to use the "wazuh-logtest" tool you can set parameters "logall" and "logall_json" to "yes" on the configuration file "ossec.conf" on the wazuh-manager.
Please take a look at these links:
https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall
https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json

After setting this parameter, please restart the wazuh-manager.

With these settings, the manager stores all collected logs in dedicated archive log files, specifically "/var/ossec/logs/archives/archives.log" and "/var/ossec/logs/archives/archives.json". These archive log files comprehensively capture all logs, including those that do not trigger any alerts.

Then, you can find there the log event you are looking for. I recommend using the "archives.json" file. And the log string that you must use for the "wazuh-logtest" tool is in the key "full_log".

Once you have the event log from the "archives.json" file, you can test it again in the "wazuh-logtest" tool to check your rule.

Regards

**Comment by Damian-Mangold** [comment]
I closed it because it had been inactive for more than 7 days. Please reopen it if necessary.
